metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala

```diff
@@ -310,19 +329,47 @@ class DestinationProvider(
       symbol: String,
       allowedBuildTargets: Set[BuildTargetIdentifier]
   ): Option[SymbolDefinition] = {
-    val definitions = index.definitions(Symbol(symbol)).filter(_.path.exists)
+    val querySymbol: Symbol = Symbol(symbol)
+    val definitions = index.definitions(querySymbol).filter(_.path.exists)
+    val extraDefinitions = if (definitions.isEmpty) {
+      // search in all classpath, i.e. jars with no source
+      buildTargets
+        .inferBuildTarget(List(querySymbol))
+        .map(f => {
+          val uri = f.jar.toURI.toString()
+          val fullURI =
+            URI.create(s"jar:${uri}!/${symbol.stripSuffix("#")}.class")
```

Comment:
This might not work for some more complex Scala symbols such as inner classes :thinking: 

In that case we might want to use ClasspathSymbols and find the class that contains the symbol.